src/VueGAPI/index.js

```diff
@@ -40,40 +40,51 @@ export default {
       })
     }
 
-    Vue.prototype.$getGapiClient = () => {
-      return new Promise((resolve, reject) => {
-        if (
-          Vue.gapiLoadClientPromise &&
-          Vue.gapiLoadClientPromise.status === 0
-        ) {
-          // promise is being executed
-          resolve(Vue.gapiLoadClientPromise)
-        } else {
-          resolveAuth2Client(resolve, reject)
-        }
-      })
+    Vue.prototype.$gapi = {
```

Comment:
This creates a problem in the future for maintainability. The question becomes do you continue to add things under the $gapi command and under there own sperate commands or do you work to move everything under $gapi?

I agree that we do not need to be polluting the Vue namespace with functions like $login and $logout but this is a major change in design and I think that @cedpoilly needs to weigh in.

Reply:
To not break existing implementations I have made this change backwards compatible. In a future release the individual instance methods should be removed. It is probably a good idea to add a deprecated warning to the code.

```suggestion
/**
 * @deprecated since version 0.0.10.
 * Will be removed in version 1.0.
 */
  Vue.prototype.$login = {
    console.warn("This global instance method is a deprecated and will be removed in a future release. Please use this.$gapi.login instead.");
    ...
}
```

Reply:
I like this warning ⚠️ idea.